org.osgi.service.feature/src/org/osgi/service/feature/FeatureService.java

```diff
@@ -33,21 +33,67 @@
      * Get a factory which can be used to build feature model entities.
      * @return A builder factory.
      */
-    public BuilderFactory getBuilderFactory();
+	BuilderFactory getBuilderFactory();
+
+	/**
+	 * Obtain an ID from a Maven ID formatted string. The supported syntax is as
+	 * follows:
+	 * <p>
+	 * {@code groupId ':' artifactId ( ':' type ( ':' classifier )? )? ':' version }
+	 * 
+	 * @param mavenID The Maven ID (not {@code null}).
+	 * @return the ID.
+	 */
+	ID getIDfromMavenID(String mavenID);
+
+	/**
+	 * Obtain an ID.
+	 * 
+	 * @param groupId The group ID (not {@code null}).
+	 * @param artifactId The artifact ID (not {@code null}).
+	 * @param version The version (not {@code null}).
+	 * @return The ID.
+	 */
+	ID getID(String groupId, String artifactId, String version);
+
+	/**
+	 * Obtain an ID.
+	 * 
+	 * @param groupId The group ID (not {@code null}).
+	 * @param artifactId The artifact ID (not {@code null}).
+	 * @param version The version (not {@code null}).
+	 * @param type The type (or {@code null}).
```

Comment:
If null, default is? `jar`?

Reply:
So empty Optional and Optional of empty string mean the same thing for classifier? And empty Optional and Optional of `jar` mean the same thing for type? Otherwise how is toString to properly render the string? You don't want `::` when classifier is an Optional of empty string.